tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/descartes/tool_pose_samplers.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file tool_pose_samplers.h
+ * @brief Tesseract tool pose sampers
+ *
+ * @author Levi Armstrong
+ * @date April 18, 2018
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2017, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_MOTION_PLANNERS_TOOL_POSE_SAMPLERS_H
+#define TESSERACT_MOTION_PLANNERS_TOOL_POSE_SAMPLERS_H
+
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <Eigen/Geometry>
+#include <type_traits>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_common/types.h>
+
+namespace tesseract_motion_planners
+{
+using ToolPoseSamplerFn = std::function<tesseract_common::VectorIsometry3d(const Eigen::Isometry3d& tool_pose)>;
+
+/**
+ * @brief Given a tool pose create samples from [-PI, PI) around the provided axis.
+ * @param tool_pose Tool pose to be sampled
+ * @param resolution The resolution to sample at
```

Comment:
I updated the documentation to state that this is the max sampling resolution, because I use ceiling to get number of samples which will insure that the actual sampling resolution is not greater than what was provided.